### PR TITLE
Adding change-detail to GET /apps response

### PR DESCRIPTION
**Pull request by sanjanasandeep**
JIRA: https://cdap.atlassian.net/browse/CDAP-19677

**Testing on my local -sandbox**
Screen Shot 2022-10-10 at 5 08 37 PM

### CODE REVIEW

**Review by sumengwang — CHANGES_REQUESTED** [review]
Can the name in the response be consistent? I believe in @GET apps/{app-id} we have `change`, but here you have `changeDetail`

**Review by de-lan — APPROVED** [review]
lgtm

**Review by sumengwang — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java [hunk 1]
```diff
@@ -2118,10 +2118,21 @@ private static final class AppScanEntry implements Map.Entry<ApplicationId, Appl
...
       this.appId = getApplicationIdFromRow(row);
       this.rawAppMeta = row.getString(StoreDefinition.AppMetadataStore.APPLICATION_DATA_FIELD);
+      String author = row.getString(StoreDefinition.AppMetadataStore.AUTHOR_FIELD);
+      String changeSummary = row.getString(StoreDefinition.AppMetadataStore.CHANGE_SUMMARY_FIELD);
+      Long creationTimeMillis = row.getLong(StoreDefinition.AppMetadataStore.CREATION_TIME_FIELD);
+      String latest = row.getString(StoreDefinition.AppMetadataStore.LATEST_FIELD);
+      if (creationTimeMillis == null) {
+        this.changeDetail = null;
+      } else {
+        this.changeDetail = new ChangeDetail(changeSummary, null, author, creationTimeMillis, latest);
```
Comment (fernst) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java:2134]: Are all of these values guaranteed to not be null?

**Review by fernst — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java (hunk 1 above)
Comment (sanjanasandeep) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java:2134]: changeSummary, author and parentVersion are nullable in ChangeDetail - only creationTime isn't, hence the additional check

**Pull request merged**